### ISSUE TITLE
Can not know staged removing files without transaction

### ISSUE THREAD

**Issue by mizchi** [isomorphic-git/isomorphic-git#284]
## tl;dr;

I can not know removing files in git without `fs` transaction clue.
isomorphic-git may need that state directly.

## Procedure

- git.init /repo
- fs.writeFile /repo/a
- git.add /repo/a
- git.commit(...) 
- fs.unlink /repo/a
- git.remove /repo/a
- git.status /repo/a => 'deleted'

In this case, if I know `/repo/a` path, I can get `/repo/a` is deleted.

- `git.listFiles` does not know /repo/a
- `fs.readdir` does not know /repo/a

So I can not reach staged removing files.

## Real World Case

I'm creating editor and git client (WIP, not OSS yet) https://nervous-kilby-73c9b0.netlify.com/

![](https://i.gyazo.com/0b3cdd87e9b5cbb4d2616f68d9724ca7.gif)

Git client can not define committable stage state.

**Comment by wmhilton** [comment]
Ah! There's been confusion on this topic before (see #206). I'm not 100% sure what you're goal is, but I think you can achieve it by calling `listFiles({ ref: 'HEAD' })`. If a file is deleted from the file system and removed from the git stage, then it is only remembered by the previous commit.

**Comment by mizchi** [comment]
> it is only remembered by the previous commit.

Ah, you are right... I can resolve this issue by `ref: 'HEAD'`

```
await git.listFiles({fs, dir: '/playground'})
(2) ["README.md"]
await git.listFiles({fs, dir: '/playground', ref: 'HEAD'})
(3) ["README.md", "a"]
```

I took `ref` was always `HEAD` for granted.

Thank you!

**Comment by wmhilton** [comment]
> I took ref was always HEAD for granted but it's not so.

Ah, shit you're right! Originally `listFiles` only listed the files in the staging area. Then in #263 I added the `ref` parameter so that listFiles could be used on commits as well. But now it is inconsistent with the rest of the API that uses a `ref` parameter that has a default value of 'HEAD'. Here the default is `undefined` which means it uses the stage... Sigh. I should start making a list of design mistakes that I can try to fix when version 1.0.0 is released.

But hey, at least it's still not as confusing as the git CLI.
